api/events/schema.py

```diff
@@ -240,20 +257,76 @@ class PaginatedDailyRoomEvents(graphene.ObjectType):
 
 
 class Query(graphene.ObjectType):
+    all_events_by_date_range = graphene.Field(
+        PaginatedEvent,
+        page=graphene.Int(),
+        per_page=graphene.Int(),
+        start_date=graphene.String(),
+        end_date=graphene.String(),
+        description="Query that returns a list of events given the arguments\
+            \n- start_date: The date and time to start selection \
+                            when filtering by time period\
+            \n- end_date: The date and time to end selection \
+                            when filtering by time period")
+
     all_events = graphene.Field(
         PaginatedDailyRoomEvents,
         start_date=graphene.String(),
         end_date=graphene.String(),
         page=graphene.Int(),
         per_page=graphene.Int(),
         description="Query that returns a list of events given the arguments\
-            \n- start_date: The date and time to start selection \
-                            when filtering by time period\
-            \n- end_date: The date and time to end selection \
-                            when filtering by time period\
+            \n- start_date: The date and time to start selection in range \
+                            when filtering by the time period\
+            \n- end_date: The date and time to end selection in range \
+                            when filtering by the time period\
             \n- page: Page number to select when paginating\
             \n- per_page: The maximum number of pages per page when paginating")
 
+    def resolve_all_events_by_date_range(self, info, **kwargs):
+        start_date = kwargs.get('start_date')
+        end_date = kwargs.get('end_date')
+        page = kwargs.get('page')
+        per_page = kwargs.get('per_page')
+        if page is not None and page < 1:
+            raise GraphQLError("page must be at least 1")
+        if per_page is not None and per_page < 1:
+            raise GraphQLError("perPage must be at least 1")
+        if page and not per_page:
+            raise GraphQLError("perPage argument missing")
+        if per_page and not page:
+            raise GraphQLError("page argument missing")
+        # get all events by date range
+        query = Events.get_query(info)
+        events = query.filter(
```

Comment:
This implementation is not efficient. You are going to be iterating through thousands of records. Move start and end date filter parameters here.

Reply:
Noted